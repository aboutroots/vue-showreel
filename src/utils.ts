export const sleep = async (ms: number): Promise<void> => {
  /**
   * Sleep for given number of miliseconds
   */
  // eslint-disable-next-line no-promise-executor-return
  await new Promise((r) => setTimeout(r, ms));
};
