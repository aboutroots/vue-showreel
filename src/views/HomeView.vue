<template>
  <section class="home container">
    <img alt="Vue logo" class="logo float" src="@/assets/img/logo.png" />
    <h1>Welcome!</h1>
    <p class="copy">
      This is an example Vue app. It's using Vue Router that allows Single Page Application
      functionality (if you start clicking on the sidebar, notice changes in the browser's search
      bar.)
    </p>
    <img src="@/assets/img/building.svg" class="bg-img" alt="Background image with building" />
  </section>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'HomeView',
});
</script>
<style lang="scss" scoped>
.home {
  width: 100%;
  text-align: center;
}
.copy {
  line-height: 1.5;
  max-width: 800px;
  margin: 0 auto;
}
</style>
