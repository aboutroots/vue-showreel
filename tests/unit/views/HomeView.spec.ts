import HomeView from '@/views/HomeView.vue';
import { shallowMount } from '../utils';

describe('HomeView tests suite', () => {
  test('renders the component successfully', () => {
    const wrapper = shallowMount(HomeView);

    const container = wrapper.find('.HomeView');

    expect(container.exists()).toBeTruthy();
  });

  test('matches snapshot', () => {
    const wrapper = shallowMount(HomeView);
    expect(wrapper.element).toMatchSnapshot();
  });
});
