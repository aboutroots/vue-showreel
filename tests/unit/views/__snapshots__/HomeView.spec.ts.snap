// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView tests suite matches snapshot 1`] = `
<section
  class="HomeView home container"
>
  <img
    alt="Vue logo"
    class="logo float"
    src="@/assets/img/logo.png"
  />
   
  <h1>
    Welcome!
  </h1>
   
  <p
    class="copy"
  >
    
    This is an example Vue app. It's using Vue Router that allows Single Page Application
    functionality (if you start clicking on the sidebar, notice changes in the browser's search
    bar.)
  
  </p>
   
  <img
    alt="Background image with building"
    class="bg-img"
    src="@/assets/img/building.svg"
  />
</section>
`;
